### DISCUSSION TITLE
are those "or" necessary?

### DISCUSSION THREAD

**Discussion by weiyuan-jiang** [GEOS-ESM/GOCART#118] [Q&A] [answered]
https://github.com/GEOS-ESM/GOCART/blob/150f6b888edef08944dcbd22fbd17d53b4c0ea79/Process_Library/Chem_MieTableMod2G.F90#L731-L736

I remove those "or"s. It seems unnecessary to me. @tclune

**Comment by tclune** [comment, marked as answer]
I think the intent is that refrIn is needed for filling in tau and ssa if they are around.

**Reply by weiyuan-jiang** [reply]
No, not here.  That is for bcsa and bext

**Comment by amdasilva** [comment]
I have not look into this in great detail, but I think I can guess the dependency. For a single species, refer and refi are intensive properties that can be pulled directly from the LUT. However, if you want a "combined index of refraction" of all aerosols you need to involve some "mixing rule" and the final result is an weighted average of all individual index of refraction, weighted by something like ssa * tau.

**Comment by amdasilva** [comment]
@pcolarco Can you take a look at this fragment:

 ```
if(present(refr) .or. present(tau) .or. present(ssa) ) then
         refrIn =   TABLE%refr(irh  ,ichannel,TYPE) * (1.-arh) &
                  + TABLE%refr(irhp1,ichannel,TYPE) * arh
      endif

      if(present(refi) .or. present(tau) .or. present(ssa) ) then
         refiIn =   TABLE%refi(irh  ,ichannel,TYPE) * (1.-arh) &
                  + TABLE%refi(irhp1,ichannel,TYPE) * arh
```

I do not understand the role of ` .or. present(tau) .or. present(ssa)` above.

**Reply by pcolarco** [reply]
What piece of code is this in?

**Reply by pcolarco** [reply]
@amdasilva This isn't in the current codebase. I see the reference now, but this is several years old and has since been refactored.
